.github/workflows/lint.yml

```
@@ -10,7 +10,7 @@ jobs:
     - name: Run GolangCI-Lint 
       uses: golangci/golangci-lint-action@v2
       with:
-        version: v1.47.0
+        version: v1.50.0
```

```suggestion
        version: latest
        only-new-issues: true
```

While we are at it, I think it is worth to mute linter on warnings not relevant to the PR and it should be fine to use it's latest version.